File: benchmarks/Java/src/cd/CallSign.java
package cd;


public class CallSign implements Comparable<CallSign> {
  private final int value;

  public CallSign(final int value) {
    this.value = value;
  }

  @Override
  public int compareTo(final CallSign other) {
    return (value == other.value) ? 0 : ((value < other.value) ? -1 : 1);
  }
}

File: benchmarks/C++HJ/cd/CallSign.cpp
#include "CallSign.h"

namespace CD {
    CallSign::CallSign(int value) {
        _value = value;
    }

    int CallSign::compareTo(shared_ptr<CallSign> other) const {
        return (_value == other->_value) ? 0 : ((_value < other->_value) ? -1 : 1);
    }
};